### PR TITLE
fix(cache): invalidate entry before calling resolver

### PR DESCRIPTION
**Pull request by clarkdo**
### 🔗 Linked issue

https://github.com/nuxt/framework/issues/5328

### ❓ Type of change

- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)

### 📚 Description

Resolves https://github.com/nuxt/framework/issues/5328

When cache is expired and multiple request come in at the same time, 1st request is pending at the handler, then 2nd request will reuse invalid response from previous expired value and 404 response may be returned.

This pr is fixing the invalid cached response by removing previous response value before handler is called.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/runtime/cache.ts [hunk 1]
```diff
@@ -52,6 +52,8 @@ export function defineCachedFunction <T=any> (fn: ((...args) => T | Promise<T>),
 
     const _resolve = async () => {
       if (!pending[key]) {
+        // Remove cached value for preventing using expired cache on concurrent requests
+        delete entry.value
```
Comment (pi0) [inline src/runtime/cache.ts:56]:
```suggestion
        entry.value = undefined
```
```suggestion
        delete entry.value
```

Comment (pi0) [inline src/runtime/cache.ts:56]:
```suggestion
        entry.value = undefined
```
```suggestion
        delete entry.value
```

Comment (clarkdo) [reply src/runtime/cache.ts]: Thanks, is this OK to be merged ?

**Review by pi0 — CHANGES_REQUESTED** [review]
Thanks <3

**Review by pi0 — APPROVED** [review]

**Pull request merged**